hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenGrantKycToAccountHandler.java

```diff
@@ -64,9 +65,8 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
         final var tokenStore = context.createStore(ReadableTokenStore.class);
         final var tokenMeta = tokenStore.getTokenMeta(op.tokenOrElse(TokenID.DEFAULT));
         if (tokenMeta == null) throw new PreCheckException(INVALID_TOKEN_ID);
-        if (tokenMeta.hasKycKey()) {
-            context.requireKey(tokenMeta.kycKey());
-        }
+        if (!tokenMeta.hasKycKey()) throw new PreCheckException(TOKEN_HAS_NO_KYC_KEY);
+        context.requireKey(tokenMeta.kycKey());
```

Comment:
Nit: there's a helper function that will do this:
```suggestion
        validateTruePreCheck(tokenMeta.hasKycKey(), TOKEN_HAS_NO_KYC_KEY);
        context.requireKey(tokenMeta.kycKeyOrThrow());   // using ...orThrow() gets rid of a null check
```
This way we match other code

Reply:
Added suggestion, although tokenmetadata record does not have kycKeyOrThrow() method, but not needed as it will fail on previous line if there is no kyc key, correct ?

Reply:
ok no worries, I can never remember which ones have `OrThrow`